src/network/networkd-manager-varlink.c

```diff
@@ -185,11 +185,17 @@ static int vl_method_set_persistent_storage(Varlink *vlink, JsonVariant *paramet
                 return r;
 
         if (ready) {
-                r = path_is_read_only_fs("/var/lib/systemd/network/");
+                _cleanup_close_ int fd = -EBADF;
+
+                fd = open("/var/lib/systemd/network/", O_CLOEXEC | O_DIRECTORY | O_PATH);
```

Comment:
xopenat() with O_DIRECTORY|O_CREAT would be cool here I think.

Reply:
networkd cannot create directory under `/var/lib/systemd/`, so `O_CREAT` cannot help us.